io/io/test/TFileMergerTests.cxx

```diff
@@ -101,3 +102,36 @@ TEST(TFileMerger, MergeSingleOnlyListed)
    ASSERT_TRUE(output.get() && output->GetListOfKeys());
    EXPECT_EQ(output->GetListOfKeys()->GetSize(), 2);
 }
+
+// https://github.com/root-project/root/issues/14558 aka https://its.cern.ch/jira/browse/ROOT-4716
+TEST(TFileMerger, MergeBranches)
+{
+   TTree atree("atree", "atitle");
+   int value;
+   atree.Branch("a", &value);
+   value = 42;
+   atree.Fill();
+
+   TTree abtree("abtree", "abtitle");
+   abtree.Branch("a", &value);
+   value = 42;
+   abtree.Fill();
+  
+   TTree dummy;
```

Comment:
This does not create a fully functional tree and thus is not representative of user code.  A fully functional tree needs a name and title.